spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -602,13 +541,21 @@ String getPullEndpoint(String subscriptionName) {
 		return this.pubSubConfiguration.computePullEndpoint(subscriptionName, projectId);
 	}
 
-	public void setThreadPoolTaskSchedulerMap(
-			ConcurrentMap<String, ThreadPoolTaskScheduler> threadPoolTaskSchedulerMap) {
-		this.threadPoolTaskSchedulerMap = threadPoolTaskSchedulerMap;
+	public void setExecutorProviderMap(
+			ConcurrentMap<String, ExecutorProvider> executorProviderMap) {
+		this.executorProviderMap = executorProviderMap;
+	}
+
+	public void setGlobalExecutorProvider(ExecutorProvider executorProvider) {
+		this.globalExecutorProvider = executorProvider;
 	}
 
-	public void setGlobalScheduler(ThreadPoolTaskScheduler threadPoolTaskScheduler) {
-		this.globalScheduler = threadPoolTaskScheduler;
+	public ExecutorProvider getGlobalExecutorProvider() {
+		return this.globalExecutorProvider;
+	}
+
+	public Map<String, ExecutorProvider> getExecutorProviderMap() {
```

Comment:
I'd rather lose the ability to validate `getExecutorProviderMap()` from autoconfigure module but keep the method package-private. Getters on collections are dangerous because collections in Java are not immutable -- anyone who calls `getExecutorProviderMap()` can modify the underlying map.

Reply:
Ah you're absolutely right. Using a mutable collection is definitely not safe and may not be worth it just for the sake of validation.